src/storage.rs

```diff
@@ -109,13 +155,26 @@ impl Storage {
 
     pub fn update_filter_scripts(&self, scripts: HashMap<Script, BlockNumber>) {
         let mut batch = self.batch();
+        // update min block number to 1 when it's setup as 0, because sync protocol will ban us when requesting genesis block data
+        let mut should_filter_genesis_block = false;
         for (script, block_number) in scripts {
-            let mut key = Key::MetaKey("FILTER_SCRIPTS").into_vec();
+            let mut key = Key::MetaKey(FILTER_SCRIPTS_KEY).into_vec();
             key.extend_from_slice(script.as_slice());
-            let value = block_number.to_be_bytes().to_vec();
+            let value = if block_number == 0 {
+                should_filter_genesis_block = true;
+                1u64.to_be_bytes().to_vec()
```

Comment:
```suggestion
                0u64.to_be_bytes().to_vec()
```
Why return `1` as value here?

Reply:
Since we need to skip the genesis block, let the filter protocol fetch the filter data starting from 1

Reply:
ok